week1/homework/index.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+let state = 10;
+const http = require('http');
+const server = http.createServer();
+
+server.on('connection', () => {
+  console.log('server is connected');
+});
+
+server.on('request', (req, res) => {
+  console.log('requesting', req.url);
+  function setResponse(header) {
+    res.setHeader('Content-Type', 'Text-html');
+    res.write(
+      `<h1 style="background-color:blue;color:white;font-size:46px;">${header}</h1>`
+    );
+    res.end();
+  }
+  switch (req.url) {
+    case '/add':
+      setResponse(state++);
```

Comment:
You are using post-increment so this will return `state` first and _then_ increment it. So even though state will end up incremented it will still pass previous state to `setResponse`. That is if initial state is 10, this will pass 10 to `setResponse` even though the new state will be 11. You should have used pre-increment instead: `++state` which will increment `state` first and then pass it to `setResponse`.